contracts/src/integrations/cert/libraries/EigenDACertVerificationLib.sol

```diff
@@ -44,19 +44,30 @@ library EigenDACertVerificationLib {
     /// @param blobQuorumsBitmap The bitmap of blob quorums
     error RequiredQuorumsNotSubset(uint256 requiredQuorumsBitmap, uint256 blobQuorumsBitmap);
 
-    /// @notice Status codes for certificate verification results
+    /// @notice Thrown when certificate decoding fails
+    error CertDecodeRevert();
+
+    /// @notice Thrown when the external call to the signature verifier reverts
+    error SignatureVerificationCallRevert(bytes reason);
+
+    /// @notice Status codes for certificate verification results.
+    /// @dev Returned by checkDACert and checkDACertV2 functions, along with an errParam bytes array for additional context.
+    ///      See the revertOnError function for what errParams contains for each error code.
     enum StatusCode {
         NULL_ERROR, // Unused error code. If this is returned, there is a bug in the code.
         SUCCESS, // Verification succeeded
         INVALID_INCLUSION_PROOF, // Merkle inclusion proof is invalid
         SECURITY_ASSUMPTIONS_NOT_MET, // Security assumptions not met
         BLOB_QUORUMS_NOT_SUBSET, // Blob quorums not a subset of confirmed quorums
-        REQUIRED_QUORUMS_NOT_SUBSET // Required quorums not a subset of blob quorums
-
+        REQUIRED_QUORUMS_NOT_SUBSET, // Required quorums not a subset of blob quorums
+        CERT_DECODE_REVERT, // Certificate abi.decoding reverted
+        SIGNATURE_VERIFICATION_CALL_REVERT // External call to signature verifier reverted
     }
 
     /// @notice Decodes a certificate from bytes to an EigenDACertV3
-    function decodeCert(bytes calldata data) internal pure returns (CT.EigenDACertV3 memory cert) {
+    /// @dev This function should not be called directly. It is exposes as external
+    //       for the purpose of try/catch'ing it inside checkDACert.
+    function _decodeCert(bytes calldata data) external pure returns (CT.EigenDACertV3 memory cert) {
```

Comment:
This should not be done because this is a library. 

The "external" tag has very different meanings in libraries vs contracts. In libraries, external means that an external library will be deployed that must be linked to the contract that is using it. This functionality is largely unsupported by many important tools, and usually results in a maintenance nightmare.

So the correct solution IMO is to define in the top level CertVerifier contract an EXTERNAL function with the same name, invoking the INTERNAL function in this library.

Reply:
ethen addressed both of these comments in [145890a](https://github.com/Layr-Labs/eigenda/pull/1926/commits/145890a243353bc1191fd664f11a4b5ffd35aaa7)